ocp4/oauth/oauth.go

```diff
@@ -16,35 +16,26 @@ func init() {
 	oauthv1.Install(scheme.Scheme)
 }
 
-// TODO: Generated yamls are results of pure imagination. Structure and consistency
-// must be reviewed and fixed. I guess this code can be simplified once we know
-// how the output should look exactly.
-
 // reference:
 //   [v3] OCPv3:
 //   - [1] https://docs.openshift.com/container-platform/3.11/install_config/configuring_authentication.html#identity_providers_master_config
 //   [v4] OCPv4:
 //   - [2] htpasswd: https://docs.openshift.com/container-platform/4.0/authentication/understanding-identity-provider.html
 //   - [3] github: https://docs.openshift.com/container-platform/4.0/authentication/identity_providers/configuring-github-identity-provider.html
 
-// Structures defining custom resource definitions / manifests / yamls
-// TODO: figure out the OKD terminology
-
 // Shared CRD part, present in all types of OAuth CRDs
 type OAuthCRD struct {
-	APIVersion string   `yaml:"apiVersion"`
-	Kind       string   `yaml:"kind"`
-	MetaData   Metadata `yaml:"metaData"`
-	Spec       struct {
+	APIVersion string `yaml:"apiVersion"`
+	Kind       string `yaml:"kind"`
+	MetaData   struct {
```

Comment:
The reason I used a separate type for Metadata is because it seems to be common practice but also we need to be able to reuse it for Secret among other things, actually Metadata is every where (or so until (and if) we have OCP4 structure from k8s libs).